designs/virtualization/4.x/expose-guest-data/expose guest data.md

```diff
@@ -0,0 +1,55 @@
+# Expose Guest Agent Data
+
+When the Guest Agent is installed on a VM some additional valuable data can be collected then surfaced in the UI.
+When the Guest Agent isn’t installed we will inform the users about that.
+
+We can report that in the VM’s Overview tab under the Status card and provide a message: “This virtual machine does not have a Guest Agent installed. Some metrics, and management features will not be available or inaccurate” and add an external link to view the details about the installation procedure of the Guest Agent, if the users are interested to do so. The link will lead the users to the instructions on how to install the Guest Agent.
+
+![Notifying users that Guest Agent isn't installed](img/vm-overview-tab.png)
+
+## After installing the Guest Agent
+
+Some of the data that is collected via the Guest Agent can be surfaced in the VM’s Overview tab under the Details card, as well as on the Details tab:
+
+- Hostname
+- Guest Agent OS version - the actual OS version that is installed on the VM
+- Time zone
+- Users list - active users that are currently logged in to the VM
+
+### VM’s Overview tab > Details card
+
+![Additional data when Guest Agent is installed](img/vm-dashboard-w-ga-installed.png)
+
+Clicking the number of logged in users will lead to a table with all the relevant info about these logged in users. This info is required so that users will be aware  when they're about to take an action that will affect those logged in users (f.e.) powering the VM off or restarting it.
+
+### Details tab
+
+![VM overview page](img/vm-overview.png)
+
+To simplify the information about OS versions (the Guest Agent and what was installed by the admin while creating the VM), we can surface only the current OS version that’s installed (the Guest Agent OS). 
+In case we have a mismatch between the chosen Operating System version (the OS that the admin originally installed) and the detected Operating System version, we can show an alert that indicates that mismatch and offer a solution.
```

Comment:
```suggestion
In case we have a mismatch between the suggested Operating System version (the OS that the admin originally suggested) and the installed Operating System version, we can show an alert that indicates that mismatch.
```
a - admin can not install an operating system.
b - we can not offer a solution, because we do not know what and if there is any problem.

Reply:
@yaacov so that means we can't provide a link that will enable the user to solve that mismatch?

Reply:
> we can't provide a link that will enable the user to solve that mismatch

I can't think of a way ...

For example, lets say an admins created a "fedora" machine ( e.g. added a label in the YAML saying they intend this machine to run fedora ) but the users need a RHEL or windows machine, so they ignore the admins suggestion and installed what they need, while taking care of all the configuration needed by themselves ... when looking at the data GA provides, we have no way to identify if everything works fine or not, and if not what is the problem.

Reply:
I guess we should omit that link then. @matthewcarleton does it make sense to just notify of a mismatch without providing any solution?

Reply:
Since the GA differs depending on the OS I'm not sure the user would ever see an error like this. 
If an admin created this VM they would have either provided the GA for Fedora ( in this example) or not at all. 
If that's true then the question is can we flag that the wrong GA is installed for a VM @yaacov ?

Otherwise we'd just state the GA is not installed and needs to be. 
That should align with the earlier designs we did [here](http://openshift.github.io/openshift-origin-design/designs/virtualization/4.3/guest-agent-not-installed/)

Reply:
a general question I have about GA for windows relates to this. Do we plan on implementing what we have for guest agent tools under the OS section [here](http://openshift.github.io/openshift-origin-design/designs/virtualization/4.3/create-vm/step-1-genaral/#operating-system-field)
If we plan to work on this it seems like a good time to do the rest of the GA related designs too :) @jelkosz WDYT?

Reply:
@yaacov Can you please explain what happens if the OS is changed but the initial GA stays as it was originally set. The wrong GA means we can’t detect the running OS. The question for that scenario is what does a GA report back if it can’t find any info?